main/params_test.go

```diff
@@ -0,0 +1,132 @@
+package main
+
+import (
+	"io/ioutil"
+	"path"
+	"testing"
+
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/ava-labs/avalanchego/chains"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/node"
+	"github.com/ava-labs/avalanchego/utils/constants"
+)
+
+func TestChainConfigs(t *testing.T) {
+	tests := map[string]struct {
+		config string
+
+		expected map[ids.ID]chains.ChainConfig
+		err      string
+	}{
+		"no chain configs": {
+			config:   `{}`,
+			expected: map[ids.ID]chains.ChainConfig{},
+		},
+		"valid chain-id (default settings)": {
+			config: `{
+				"chain-configs": [
+					{"chain-id": "yH8D7ThNJkxmtkuv2jgBa4P1Rn3Qpr4pPr7QYNfcdoS6k6HWp", "settings": "default", "upgrades": "default"},
+					{"chain-id": "2JVSBoinj9C2J33VntvzYtVJNZdN2NKiwwKjcumHUWEb5DbBrm", "settings": "blah1", "upgrades": "blah2"}
+				]
+			}`,
+			expected: func() map[ids.ID]chains.ChainConfig {
+				m := map[ids.ID]chains.ChainConfig{}
+				id1, err := ids.FromString("yH8D7ThNJkxmtkuv2jgBa4P1Rn3Qpr4pPr7QYNfcdoS6k6HWp")
+				assert.NoError(t, err)
+				m[id1] = chains.ChainConfig{Settings: []byte("default"), Upgrades: []byte("default")}
```

Comment:
Would it be possible to change the behavior from passing in `[]byte("default")` to passing in nil instead?

Reply:
As you can probably tell, not populating a given key will result in the field being `nil`. I'll add a test that demonstrates this behavior.